components/ShoppingListEditor/ShoppingListEditor.js

```diff
@@ -2,55 +2,99 @@ import styled from "styled-components";
+  useEffect(() => {
+    triggerDropDown();
+  }, [itemInput]);
 
-    const trimmedItemName = itemName.trim(); //remove white spaces at beginning and end of the name
+  //reset dropDownItems state, to remove dropDown buttons from screen,...
+  //...when shopping list (items) changed
+  useEffect(() => {
+    setItemInput("");
+    setDropDownItems([]);
+  }, [items]);
 
-    // --- Check user input ---
-    if (trimmedItemName === "") {
-      alert("Type an item name!");
-      return;
+  //...when loosing focus of input field
+  //-> setDropDownReset state needed to allow clicking drop down buttons,
+  //which also removes focus on input field
+  useEffect(() => {
+    if (dropDownReset) {
+      setDropDownItems([]);
+      setDropDownReset(false);
     }
-    if (
-      items.find(
-        (item) => item.name.toLowerCase() === trimmedItemName.toLowerCase()
-      ) !== undefined
-    ) {
-      alert("Item already on the list!");
-      return;
+  }, [dropDownReset]);
+
+  //evoke rendering drop down buttons for items that match input
+  function triggerDropDown() {
+    const matchedDropDownItems = matchInput(itemInput);
+    setDropDownItems(matchedDropDownItems);
+  }
+
+  //match input with all available items from DB
+  function matchInput(value) {
+    const editedValue = value.trim().toLowerCase(); //trim() -> remove white spaces at beginning and end of the name
+    //clear drop down when clearing input field
+    if (editedValue === "") {
+      return [];
     }
+    const matchedItems = allItems.filter((item) =>
+      item.name.toLowerCase().startsWith(editedValue)
+    );
+    //filter out matched items, that are already on the shopping list
+    const uniqueMatchedItems = matchedItems.filter((matchedItem) => {
+      return (
+        items.find((item) => matchedItem.id === item.item_id) === undefined
+      );
+    });
+    return uniqueMatchedItems;
+  }
 
-    onAdd(trimmedItemName);
-    form.reset();
-    inputRef.current.focus(); //set focus on input field
+  function onSubmit(event) {
+    event.preventDefault();
+    event.target.reset();
+    inputRef.current.focus();
   }
 
   return (
     <ListContainer>
-      <StyledForm aria-label="add items" onSubmit={submitForm}>
+      <StyledForm
+        aria-label="add items"
+        onSubmit={(event) => onSubmit(event)}
+        autoComplete="off"
+      >
         <label htmlFor="item">Item</label>
         <input
           type="text"
-          name="item"
           id="item"
           aria-label="item name"
           placeholder="Brot"
           maxLength="30"
-          ref={inputRef} // set ref to set autofocus after submit
+          ref={inputRef} //set ref to set autofocus after submit
+          value={itemInput}
+          onInput={(event) => setItemInput(event.target.value)} //don't use onChange() -> it ignores some events!!!
+          onFocus={triggerDropDown}
+          onBlur={() => setDropDownReset(true)} //to close drop down, when losing focus
```

Comment:
There is a simpler way of displaying the dropdown. use a state with all the dropdownitems, starting with an empty array. Then use conditional rendering which disables the dropdown if the array is empty (something like 
`{dropdownItems.length <=0 && ...}`. Then search for matching items every time the input value changes. On a click on the add button of an dropDownItem you could either clear the form input which or set the dropDownItems to an empty array, both resulting in hiding the dropdown list.

Reply:
You still need the onBlur function, but here you can simply reset the input and clear the dropdownItems array